src/runtime/hexagon/sim/hexagon_device_sim.cc

```diff
@@ -84,6 +82,34 @@ std::unique_ptr<T> make_unique(size_t size) {
   return std::unique_ptr<T>(new U[size]());
 }
 
+// Replacement for llvm::Optional.
+template <typename T>
+class Optional {
+ public:
+  Optional() : has_value(false) {}
+  Optional(const T& val) : value(val), has_value(true) {}  // NOLINT(*)
+  Optional(T&& val) : value(val), has_value(true) {}       // NOLINT(*)
+
+  bool hasValue() const { return has_value; }
+  operator bool() const { return hasValue(); }
+  T operator*() const {
+    ICHECK(has_value) << "value not set";
+    return value;
+  }
+  const T* operator->() const {
+    ICHECK(has_value) << "value not set";
+    return &value;
+  }
+  T* operator->() {
+    ICHECK(has_value) << "value not set";
+    return &value;
+  }
+
+ private:
+  T value;
+  bool has_value;
+};
+
```

Comment:
dmlc has an optional type, maybe use that instead?

Reply:
Done.